src/Avalonia.Visuals/Media/Imaging/RenderTargetBitmap.cs

```diff
@@ -21,21 +23,27 @@ public class RenderTargetBitmap : Bitmap, IDisposable, IRenderTarget
         /// <param name="dpiX">The horizontal DPI of the bitmap.</param>
         /// <param name="dpiY">The vertical DPI of the bitmap.</param>
         public RenderTargetBitmap(int pixelWidth, int pixelHeight, double dpiX = 96, double dpiY = 96)
-            : base(CreateImpl(pixelWidth, pixelHeight, dpiX, dpiY))
+           : this(RefCountable.Create(CreateImpl(pixelWidth, pixelHeight, dpiX, dpiY)))
         {
         }
 
+        private RenderTargetBitmap(IRef<IRenderTargetBitmapImpl> impl) : base(impl)
```

Comment:
Why is this private constructor needed? It's only called by the public constructor - why not just put this in the public constructor?

Reply:
You see, that's the limitation of C#. You can't create a new object in constructor and pass it to base constructor. So if you want to pass something to base constructor *and* save it to a field, you need a separate private constructor.